lib/vrl/stdlib/src/compact.rs

```diff
@@ -336,11 +336,16 @@ mod test {
                 Default::default(),
             ),
             (
-                vec![1.into(), Value::Object(map!["field2": 2]), 2.into()],
                 vec![
                     1.into(),
-                    Value::Object(map!["field1": Value::Null,
-                                    "field2": 2]),
+                    Value::Object(BTreeMap::from([(String::from("field2"), Value::from(2))])),
+                ],
+                vec![
+                    1.into(),
+                    Value::Object(BTreeMap::from([
+                        (String::from("field1"), Value::Null),
+                        (String::from("field2"), Value::from(2)),
+                    ])),
                     2.into(),
```

Comment:
nit: given that `BTreeMap::from` is already fairly verbose, I usually do `"field2".into()` and `2.into()`. Given that these are exactly known to be `String` and `Value`, there should be no ambiguity here.

Reply:
That's a good point. As a matter of preference I prefer, generally, the `from` variant. I find it a little easier to read -- I don't have a good memory for what type a thing is going into unless it's explicit -- and the search and replace aspect is much simpler. Very much a quirk of mine.